Add a python mockup based on QTableView: supports copy/pasting to/from excel, undo operations

jmarrec opened the pull request:
![python_mockup](https://github.com/jmarrec/QtTestbed/assets/5479063/7166b036-312a-4229-b38d-5ecb8ec387e7)


Here on Ubuntu, show the copy/paste to/from a spreadsheet editor:

![QtableView_copy_paste_excel](https://github.com/jmarrec/QtTestbed/assets/5479063/31cc297e-4107-4f85-a11f-ee6f09de0ac0)

jmarrec committed:
Add a python mockup based on QTableView: supports copy/pasting to/from excel, undo operations

macumber commented on ModelDesignWizard_python_QTableView/ModelDesignWizard.py:
```diff
@@ -0,0 +1,590 @@
+# Copyright (C) 2022 The Qt Company Ltd.
+# SPDX-License-Identifier: LicenseRef-Qt-Commercial OR BSD-3-Clause
+
+import csv
```
Woah, this is cool.  What would this require to run?  Could it be run with the openstudio cli or require a different python interpreter?   Would it load the Qt dlls we ship with the app?  I definitely think a python or ruby wizard UI would be preferable to a Qt one for wizards in terms of people making their own custom wizards

jmarrec replied:
I think you can mix c++ and python qt elements. You're supposed to embbed a python interpreter in your c++ app I believe

jmarrec replied:
Here I was just trying to prototype faster (and try pyside6 which I hadn't yet).
But I see your point about being able to add your own wizards as an advanced user... interesting!

jmarrec replied:
I guess I'll try to call the openstudio CLI on it, provided a custom --python_path arg where I pip installed pyside6 and all.

macumber replied:
Ok cool, I was just trying to see where you were going with this.  QML/QtQuick might also be interesting to try at some point, it would let the UI be packaged with the measure and loaded at run time instead of compiled in the app.  Not trying to stop what you're doing, just throwing out another idea. - either python or QML could have the advantage of allowing user defined UIs for wizards that get loaded at runtime

jmarrec replied:
This seems to work (locale aside)

```
$os_build_rel/Products/openstudio labs --python_path /home/julien/Virtualenvs/qt/lib/python3.8/site-packages ModelDesignWizard.py
```


Not sure that's really what we would want though, because it poses the problem of passing info from C++ to python and back... In this specific case, it's kinda fine because we could pass the link to the temp dir to python, and python could prepare the workflow.json (and call the CLI on it or not (C++ would do it)), but otherwise we'd always have to serialize the model (save) and reload when switching from C++ to python

macumber commented on ModelDesignWizard_python_QTableView/ModelDesignWizard.json:
Is this file exported from OpenStudio Standards or something?

jmarrec replied:
These are harcoded into a few functions in a utilility module of the ruby measure. I have a ruby script that calls it and prepares that json. It's on an unpushed branch of OSapp

macumber replied:
Ok cool, and then once all the inputs are selected would the wizard call the measure to create the model?

jmarrec replied:
It'd write the workflow.json with the right inputs at least, but probably it would call the measure too (syscall to OS CLI). We may want to run the "create bar" measure first, then in a second pass use the "create typical model" measure.

jmarrec committed:
Make it work with python 3.8, which the openstudio CLI uses

```
$os_build_rel/Products/openstudio labs --python_path /home/julien/Virtualenvs/qt/lib/python3.8/site-packages ModelDesignWizard.py
```
